pkg/registry/servicecatalog/broker/storage.go

```diff
@@ -143,5 +145,27 @@ func NewStorage(opts server.Options) (brokers, brokersStatus rest.Storage) {
 	statusStore := store
 	statusStore.UpdateStrategy = brokerStatusUpdateStrategy
 
-	return &store, &statusStore
+	return &store, &StatusREST{&statusStore}
+}
+
+// StatusREST defines the REST operations for the status subresource.
+type StatusREST struct {
+	store *registry.Store
+}
+
+// New returns a new ServiceBroker.
+func (r *StatusREST) New() runtime.Object {
+	return &servicecatalog.ServiceBroker{}
```

Comment:
Could use EmptyObject() here.

Reply:
Actually, it doesn't look like this is needed since NewFunc is called for New.

Reply:
Turns out it is needed, rest.Storage requires it.

Reply:
ok, I'll ask the dumb question :-)   why does a New() on StatusREST return something other than a StatusREST struct?  Isn't that kind of misleading/confusing?

Reply:
Oh never mind I think I get it now, however why isn't this returning ServiceBrokerStatus{} instead? Or does the code that acts on the returned `brokersStatus rest.Storage` assume its actually the full object and will look for a `Status` field in there?